unit-tests/src/test/java/org/eclipse/collections/impl/map/mutable/UnifiedMapTest.java

```diff
@@ -866,6 +866,16 @@ public void trimToSize()
         Assert.assertEquals(5, map.size());
     }
 
+    @Test
+    public void merge()
+    {
+        MutableMap<Integer, Integer> actualMap = this.mapWithCollisionsOfSize(8);
+        Integer merge = actualMap.merge(COLLISION_1, 1, (integer1, integer2) -> null);
+        Assert.assertNull(merge);
+        MutableMap<Integer, Integer> expectedMap = MORE_COLLISIONS.subList(1, 8).toMap(k -> k, v -> v);
+        Assert.assertEquals(expectedMap, actualMap);
+    }
```

Comment:
I stepped through this test in a debugger and it hits the line `this.overwriteWithLastElementFromChain(chain, index, i)`